unison-runtime/src/Unison/Runtime/Builtin.hs

```diff
@@ -1379,15 +1306,16 @@ outMaybeNat tag result n =
       [ (0, ([], none)),
         ( 1,
           ( [UN],
+          -- TODO: Fix this?
             TAbs result
               . TLetD n BX (TCon Ty.natRef 0 [n])
               $ some n
```

Comment:
When I tried the "simple fix" here it broke, I think I'm missing something and need to revisit this.

Reply:
I don't know what you tried, but the original code doesn't look like it makes sense.

Note that the same variable, `n`, is used for the boxed nat _and_ the unboxed nat that's going inside it, and `result` is never used. Presumably `result` is the actual unboxed nat, so the new version should be modified in that light.

Reply:
Yeah I was very confused haha, I didn't think the original made sense, but changing it broke tests, so it's apparently being used _somehow_

Reply:
I changed it to `TCon Ty.natRef 0 [result]` locally and it seemed to work. But maybe I wasn't running the right test. It looks like it's only used for the process exit code return.

But yeah, you're right that there doesn't even need to be a special numeric version of this anymore.